ppb_vector/vector2.py

```diff
@@ -51,6 +51,46 @@ def _find_lowest_vector(left: typing.Type, right: typing.Type) -> typing.Type:
 
 @dataclass(eq=False, frozen=True, init=False, repr=False)
 class Vector2:
+    """The immutable, 2D vector class of the PursuedPyBear project.
+
+    :py:class:`Vector2` is an immutable 2D Vector, which can be instantiated as
+    expected:
+
+        >>> from ppb_vector import Vector2
+        >>> Vector2(3, 4)
+        Vector2(3.0, 4.0)
+
+    :py:class:`Vector2` implements many convenience features, as well as
+    useful mathematical operations for 2D geometry and linear algebra.
+
+    :py:class:`Vector2` implements :py:class:`Sequence
+    <collections.abc.Sequence>` (and subsequently :py:class:`Iterable
+    <collections.abc.Iterable>`), allowing usage like converting, indexing, and unpacking:
```

Comment:
I left these off deliberately, especially `Sequence` because `Vector2` is not technically a `collections.abc.Sequence` because it does not implement several of the functions.

It's just an informal sequence because it's iterable, indexable, and has a length.

Reply:
Ah, ok.

Reply:
@astronouth7303 BTW, why aren't we inheriting from `collections.abc.Sequence`?

Reply:
I don't remember if there was a particular reason, or just "we only kinda pretend to be a sequence, but it's more like a struct/bag", or just didn't bother?

Reply:
Should we inherit from it?  Being “kinda sort a sequence but not really” sounds like there are some fun edge-cases waiting to bite someone in the butt.

Reply:
(Definitely not something for this PR though)

Reply:
The unimplemented stuff is kinda weird on a Vector? Especially the search stuff.

Reply:
Like, if you look at the [methods associated with `Sequence`](https://docs.python.org/3/library/collections.abc.html), many of them vary from "ehhhhh" to "just kinda not applicable?".